docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,205 @@
+# ImageSegmentation.jl
+
+## Introduction
+
+Image Segmentation is the process of partitioning the image into regions that have similar attributes. Image segmentation has various applications e.g, medical image segmentation, image compression and is used as a preprocessing step in higher level vision tasks like object detection and optical flow. This package is a collection of image segmentation algorithms written in Julia.
+
+## Installation
+
+```julia
+Pkg.add("ImageSegmentation")
+```
+
+## Functions
+
+All the segmentation algorithms (except Fuzzy C-means) return a struct `SegmentedImage`
+that captures most of the details regarding the segments.
+
+* `image_indexmap`      : An array conatining the assigned labels for each pixel
+* `segment_labels`      : List of all the applied labels
+* `segment_means`       : Dict(Label => Mean intensity)
+* `segment_pixel_count` : Dict(Label => Pixel Count)
+
+#### Seeded Region Growing
+
+This algorithm segments an image with repsect to a set of *n* seeds. Given the
+set of seeds in the form of a vector of tuples of `CartesianIndex` and label,
+the algorithm tries to assign these labels to each of the remaining points.
+
+###### Inputs
+
+* `img`             :  N-D image to be segmented (arbitrary indices are allowed)
```

Comment:
This kind of detailed information about the arguments could go in `function_reference.md`, and be taken directly from the docstring. I'd rather make this section more expository.